inject/src/main/java/io/micronaut/inject/visitor/VisitorContext.java

```diff
@@ -50,10 +52,18 @@ public interface VisitorContext extends MutableConvertibleValues<Object>, ClassW
 
     /**
      * Gets the element factory for this visitor context.
+     *
      * @return The element factory
      * @since 2.3.0
      */
-    @NonNull ElementFactory<?, ?, ?, ?> getElementFactory();
+    @NonNull
+    ElementFactory<?, ?, ?, ?> getElementFactory();
+
+    @NonNull
+    ElementAnnotationMetadataFactory getElementAnnotationMetadataFactory();
+
+    @NonNull
+    AbstractAnnotationMetadataBuilder<?, ?> getAnnotationMetadataBuilder();
```

Comment:
can we not make this public API?

Reply:
I need it for one use-case that I can rewrite in the next PR. I have added `@Internal` for now.